### PR TITLE
Support platform specific signals on Linux, AIX and MacOS

### LINKED ISSUE #6967: Platform specific signals supported by Hotspot but not by OpenJ9
### Summary

`TestSignalSupport` was run on Linux, AIX, MacOS and Windows. Comparing the test output between Hotspot and OpenJ9, it was found that the following platform specific signals are supported by Hotspot but not by OpenJ9:

- **Linux:** `SIGCLD`, `SIGPWR`, `SIGSTKFLT`.
- **AIX:** `SIGCLD`, `SIGCPUFAIL`, `SIGDANGER`, `SIGEMT`, `SIGGRANT`, `SIGLOST`, `SIGMIGRATE`, `SIGMSG`, `SIGPRE`, `SIGPWR`, `SIGRETRACT`, `SIGSAK`, `SIGSOUND`.
- **MacOS:** `SIGEMT`.
- **Windows:** None.

### TestSignalSupport.java
```
import sun.misc.Signal;

public class TestSignalSupport {
	private static String[] signals = {"ABRT", "ALRM", "BREAK", "BUS", "CHLD", "CLD", "CONT", "CPUFAIL", "DANGER", "EMT", "FPE", "FREEZE", "GRANT", "HUP", "ILL", "INFO", "INT", "IO", "IOT", "KILL", "LOST", "LWP", "MIGRATE", "MSG", "PIPE", "POLL", "PRE", "PROF", "PWR", "QUIT", "RECONFIG", "RETRACT", "SAK", "SEGV", "SOUND", "STKFLT", "STOP", "SYS", "TERM", "THAW", "TRAP", "TSTP", "TTIN", "TTOU", "UNUSED", "URG", "USR1", "USR2", "VTALRM", "WATIING", "WINCH", "XCPU", "XFSZ", "XRES", "RTMIN", "RTMAX"};

	public static void main(String args[]) {
   		for (String signal : signals) {
			try {
				Signal sig = new Signal(signal);
				System.out.println("Supported signal: SIG" + signal);
			} catch (IllegalArgumentException e) {
				System.out.println("Unsupported signal: SIG" + signal);
			}
		}
	}
}
```

### Test Output

| Operating System | Hotspot Test Output | OpenJ9 Test Output |
| ------------- | ------------- | ------------- |
| Linux | [hotspot_linux_signals.txt](https://github.com/eclipse/openj9/files/3575759/hotspot_linux_signals.txt) | [openj9_linux_signals.txt](https://github.com/eclipse/openj9/files/3575763/openj9_linux_signals.txt) |
| AIX | [hotspot_aix_signals.txt](https://github.com/eclipse/openj9/files/3575758/hotspot_aix_signals.txt) | [openj9_aix_signals.txt](https://github.com/eclipse/openj9/files/3575762/openj9_aix_signals.txt) |
| MacOS | [hotspot_mac_signals.txt](https://github.com/eclipse/openj9/files/3575760/hotspot_mac_signals.txt) | [openj9_mac_signals.txt](https://github.com/eclipse/openj9/files/3575764/openj9_mac_signals.txt) |
| Windows | [hotspot_win64_signals.txt](https://github.com/eclipse/openj9/files/3575761/hotspot_win64_signals.txt) | [openj9_win64_signals.txt](https://github.com/eclipse/openj9/files/3575765/openj9_win64_signals.txt) |

### PR DESCRIPTION
**Pull request by babsingh**
The following platform specific signals are now supported by OpenJ9:

- Linux: SIGCLD, SIGPWR, SIGSTKFLT.
- AIX: SIGCLD, SIGCPUFAIL, SIGDANGER, SIGEMT, SIGGRANT, SIGLOST,
SIGMIGRATE, SIGMSG, SIGPRE, SIGPWR, SIGRETRACT, SIGSAK, SIGSOUND.
- MacOS: SIGEMT.

Closes #6967.
Dependent on https://github.com/eclipse/omr/pull/4407.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

File: runtime/j9vm/jvm.c [hunk 1]
```diff
@@ -450,6 +450,48 @@ static const J9SignalMapping signalMap[] = {
 #if defined(SIGPOLL)
 	J9_SIGNAL_MAP_ENTRY("POLL", SIGPOLL),
 #endif /* defined(SIGPOLL) */
+#if defined(SIGPWR)
```
Comment (DanHeidinga) [inline runtime/j9vm/jvm.c]: Is this supposed to be 
```suggestion
#if defined(SIGCLD)
```
?

File: runtime/j9vm/jvm.c [hunk 2]
```diff
@@ -450,6 +450,48 @@ static const J9SignalMapping signalMap[] = {
...
+	J9_SIGNAL_MAP_ENTRY("CLD", SIGCLD),
+#endif /* defined(SIGPWR) */
```
Comment (DanHeidinga) [inline runtime/j9vm/jvm.c]:
```suggestion
#endif /* defined(SIGCLD) */
```

Comment (DanHeidinga) [reply runtime/j9vm/jvm.c]: If it's supposed to be `SIGPWR, the two sets of ifdefs should be merged

Comment (babsingh) [reply runtime/j9vm/jvm.c]: it should be `SIGCLD`; the MAP_ENTRY should be wrapped using the associated signal flag.

**Pull request closed without merging**